### ISSUE TITLE
Pycharm and Intellij-idea signature have been modified after installed

### ISSUE THREAD

**Issue by arcticlion** [caskroom/homebrew-cask#9572]
OS is OS X 10.10.2
```
codesign -vvv /opt/homebrew-cask/Caskroom/pycharm/4.0.4/PyCharm.app/
/opt/homebrew-cask/Caskroom/pycharm/4.0.4/PyCharm.app/: invalid Info.plist (plist or signature have been modified)
In architecture: x86_64
```
```
/opt/homebrew-cask/Caskroom/intellij-idea/14.0.3/IntelliJ IDEA 14.app/: invalid Info.plist (plist or signature have been modified)
In architecture: x86_64
```

**Comment by radeksimko** [comment]
Hi,
this is gonna be most likely my fault.

There's generally an issue with all IntelliJ IDEs and **hard-coded** Java 6. I tried to make it work under newer Java, but it seems that even though these IDEs work, upstream (JetBrains) [says it's not supported](https://github.com/caskroom/homebrew-cask/issues/6910#issuecomment-60113422).

The hack which allows IDEs to run under Java 6+ needs to be replaced with `caveat` and later on with [a proper dependency stanza](https://github.com/caskroom/homebrew-cask/issues/7993).

**Comment by arcticlion** [comment]
@radeksimko I meet a problem about IntelliJ IDEs using Java7, it will case memory leak and Java7 doesn't support Mac font rendering technology.

Then what about Pycharm?

**Comment by radeksimko** [comment]
PyCharm is IntelliJ-based IDE too.

**Comment by w7tek** [comment]
@radeksimko I believe that there is a solution that avoids triggering the OS's "modified after installed" warning for the application. According to [this youtrack issue at JetBrains] (https://youtrack.jetbrains.com/issue/IDEA-126847), the preferred method is to change the 1.6* in idea.properties and not in Info.plist.

Unfortunately, according to the conversation along the same issue, it appears the idea.properties is picked up from a different location when using the CE than when using Ultimate.

**Comment by radeksimko** [comment]
Fixed via #9943
